caas/kubernetes/provider/storage.go

```diff
@@ -17,52 +17,46 @@ import (
 	"github.com/juju/juju/caas/kubernetes/provider/storage"
 	jujucontext "github.com/juju/juju/environs/context"
 	jujustorage "github.com/juju/juju/storage"
-	storageprovider "github.com/juju/juju/storage/provider"
 )
 
+func validateStorageAttributes(attributes map[string]interface{}) error {
+	if _, err := storage.ParseStorageConfig(attributes); err != nil {
+		return errors.Trace(err)
+	}
+	if _, err := storage.ParseStorageMode(attributes); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+type storageProvider struct {
+	client *kubernetesClient
+}
+
+var _ jujustorage.Provider = (*storageProvider)(nil)
+
 //ValidateStorageProvider returns an error if the storage type and config is not valid
 // for a Kubernetes deployment.
-func ValidateStorageProvider(providerType jujustorage.ProviderType, attributes map[string]interface{}) error {
-	switch providerType {
-	case constants.StorageProviderType:
-	case storageprovider.RootfsProviderType:
-	case storageprovider.TmpfsProviderType:
-	default:
-		return errors.NotValidf("storage provider type %q", providerType)
-	}
+func (g *storageProvider) ValidateForK8s(attributes map[string]any) error {
```

Comment:
Why do we need to 2 validate methods on the storage providers?  Especially if one is specific only k8s and the other is a subset of it.

Reply:
It's good to pull it out. Can we rename the method to Validate? Rather than have a specific K8s method?

Reply:
In the original code there was a distinction between configuration for k8s and other scenarios. This maintains that distinction by forcing every provider to say whether they can work for K8s with a given configuration or not.

Reply:
I disagree that is necessary. When a generic name, providers can do whatever validation they require, if they require it.

Reply:
In the former code there was a specific logic differentiating between VMs and containers. This enforces storage providers to implement a specific function for K8s deployments. This is following the previous logic by clearly saying that this is only for K8s.